SimCalorimetry/EcalEBTrigPrimAlgos/src/EcalEBPhase2TrigPrimAlgo.cc

```diff
@@ -0,0 +1,299 @@
+#include <string>
+#include <algorithm>
+#include <numeric>
+#include <functional>
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+//#include "Geometry/CaloGeometry/interface/CaloGeometry.h"
+//#include "Geometry/Records/interface/CaloGeometryRecord.h"
+#include "Geometry/EcalMapping/interface/EcalElectronicsMapping.h"
+#include "Geometry/EcalMapping/interface/EcalMappingRcd.h"
+#include "Geometry/EcalAlgo/interface/EcalBarrelGeometry.h"
+
+#include "SimCalorimetry/EcalEBTrigPrimAlgos/interface/EcalEBPhase2TrigPrimAlgo.h"
+
+#include "DataFormats/EcalDetId/interface/EcalSubdetector.h"
+#include "DataFormats/EcalDigi/interface/EBDataFrame_Ph2.h"
+#include "DataFormats/EcalDetId/interface/EcalTrigTowerDetId.h"
+#include "DataFormats/EcalDetId/interface/EBDetId.h"
+#include "DataFormats/EcalDetId/interface/EcalTriggerElectronicsId.h"
+
+#include "CondFormats/EcalObjects/interface/EcalTPGPedestals.h"
+#include "CondFormats/DataRecord/interface/EcalTPGPedestalsRcd.h"
+
+#include "DataFormats/EcalDigi/interface/EcalConstants.h"
+
+#include <TTree.h>
+#include <TMath.h>
+
+//----------------------------------------------------------------------
+
+const unsigned int EcalEBPhase2TrigPrimAlgo::nrSamples_ = ecalPh2::sampleSize;  // 16 samples
```

Comment:
Rather funny the usage of `nrSamples_` and `maxNrSamples_`: why not to use `ecalPh2::sampleSize` for all?

Reply:
No, because nrSamples (not set to 16) might become smaller at a later stage since we will almost certainly noy want to make a  finale TP of 16 samples. For example in PhaseI, that number is 5, i.e. BX0+-2